### PR TITLE
fix: adds pyodbc output converter for DATETIMEOFFSET

### LINKED ISSUE #253: Cannot use `dbt source freshness` with DATETIMEOFFSET column types
Running into issues with [`dbt source freshness`](https://docs.getdbt.com/reference/resource-properties/freshness) pointing at DATETIMEOFFSET columns, with pyodbc throwing the following error:

````
ODBC data type -155 is not supported.
````

Directly relates to https://github.com/mkleehammer/pyodbc/issues/134

### PR DESCRIPTION
**Pull request by baldwicc**
Fixes #253

Refs:
https://github.com/mkleehammer/pyodbc/wiki/Using-an-Output-Converter-function
https://github.com/mkleehammer/pyodbc/issues/134#issuecomment-281739794

### CODE REVIEW

File: dbt/adapters/sqlserver/connections.py [hunk 1]
```diff
@@ -271,6 +272,40 @@ def get_pyodbc_attrs_before(credentials: SQLServerCredentials) -> Dict:
     return attrs_before
 
 
+def handle_datetimeoffset(dto_value) -> datetime:
```
Comment (sdebruyn) [inline dbt/adapters/sqlserver/connections.py:275]: Would it be possible to add a unit test for this function?

File: dbt/adapters/sqlserver/connections.py [hunk 2]
```diff
@@ -271,6 +272,40 @@ def get_pyodbc_attrs_before(credentials: SQLServerCredentials) -> Dict:
...
+    Source
+    ------
+    pyodbc wiki:
+        https://github.com/mkleehammer/pyodbc/wiki/Using-an-Output-Converter-function
+
+    @stevenwinfield's decoding suggestion here:
+        https://github.com/mkleehammer/pyodbc/issues/134#issuecomment-281739794
+    """
+    tup = struct.unpack("<6hI2h", dto_value)  # e.g., (2017, 3, 16, 10, 35, 18, 500000000, -6, 0)
+    return datetime(
```
Comment (sdebruyn) [inline dbt/adapters/sqlserver/connections.py:297]: Named arguments would make this code a little bit clearer

File: dbt/adapters/sqlserver/connections.py (hunk 1 above)
Comment (baldwicc) [reply dbt/adapters/sqlserver/connections.py:275]: I've added a broader functional test for `dbt source freshness` for different types so this gets coverage, but haven't done a specific test for this handler logic.

File: dbt/adapters/sqlserver/connections.py (hunk 2 above)
Comment (baldwicc) [reply dbt/adapters/sqlserver/connections.py:299]: Done

**Comment by matsonj** [comment]
Updated with this PR - this fixes the failing tests.

https://github.com/baldwicc/dbt-sqlserver/pull/7

**Comment by baldwicc** [comment]
Done - saw this has v1.1.1 milestone so updated changelog.md accordingly

**Pull request closed without merging**